kernel/kernel-api/src/test/scala/io/delta/kernel/internal/catalogManaged/SnapshotBuilderSuite.scala

```diff
@@ -278,5 +278,190 @@ class SnapshotBuilderSuite extends AnyFunSuite
+        .withLogData(parsedRatifiedStagedCommits(Seq(0, 1, 2)).toList.asJava)
+        .withMaxCatalogVersion(5)
+        .build(emptyMockEngine)
+    }.getMessage
+
+    assert(exMsg === "Provided catalog commits must end with max catalog version")
+  }
+
+  test("withMaxCatalogVersion: without version, logData ending with maxCatalogVersion is valid") {
+    // Should not throw - validation passes
+    TableManager.loadSnapshot(dataPath.toString)
+      .asInstanceOf[SnapshotBuilderImpl]
+      .withProtocolAndMetadata(protocolWithCatalogManagedSupport, metadata)
+      .withLogData(parsedRatifiedStagedCommits(Seq(0, 1, 2, 3, 4, 5)).toList.asJava)
+      .withMaxCatalogVersion(5)
+      .build(emptyMockEngine)
+  }
+
+  test("withMaxCatalogVersion: empty logData with maxCatalogVersion is valid") {
+    // Should not throw - empty logData is allowed
+    TableManager.loadSnapshot(dataPath.toString)
+      .asInstanceOf[SnapshotBuilderImpl]
+      .atVersion(5)
+      .withProtocolAndMetadata(protocolWithCatalogManagedSupport, metadata)
+      .withLogData(Collections.emptyList())
+      .withMaxCatalogVersion(5)
+      .build(emptyMockEngine)
+  }
+
+  test("withMaxCatalogVersion: version time-travel with logData not including version fails") {
+    val exMsg = intercept[IllegalArgumentException] {
+      TableManager.loadSnapshot(dataPath.toString)
+        .atVersion(5)
+        .withLogData(parsedRatifiedStagedCommits(Seq(0, 1, 2, 3)).toList.asJava)
+        .withMaxCatalogVersion(10)
+        .build(emptyMockEngine)
+    }.getMessage
+
+    assert(exMsg === "Provided catalog commits must include versionToLoad for time-travel queries")
+  }
+
+  test("withMaxCatalogVersion: version time-travel with logData ending at version is valid") {
+    // Should not throw - logData ends exactly at requested version
+    TableManager.loadSnapshot(dataPath.toString)
+      .asInstanceOf[SnapshotBuilderImpl]
+      .atVersion(5)
+      .withProtocolAndMetadata(protocolWithCatalogManagedSupport, metadata)
+      .withLogData(parsedRatifiedStagedCommits(Seq(0, 1, 2, 3, 4, 5)).toList.asJava)
+      .withMaxCatalogVersion(10)
+      .build(emptyMockEngine)
+  }
+
+  test("withMaxCatalogVersion: version time-travel with logData beyond version is valid") {
+    // Should not throw - logData extends beyond requested version
+    TableManager.loadSnapshot(dataPath.toString)
+      .asInstanceOf[SnapshotBuilderImpl]
+      .atVersion(5)
+      .withProtocolAndMetadata(protocolWithCatalogManagedSupport, metadata)
+      .withLogData(parsedRatifiedStagedCommits(Seq(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10)).toList.asJava)
+      .withMaxCatalogVersion(10)
+      .build(emptyMockEngine)
+  }
+
+  test("validateMaxCatalogVersionPresence: catalogManaged table requires maxCatalogVersion") {
+    val exMsg = intercept[IllegalArgumentException] {
+      TableManager.loadSnapshot(dataPath.toString)
+        .asInstanceOf[SnapshotBuilderImpl]
```

Comment:
do we need this cast btw? are we using any internal or non-public APIs?

Looking at `.atVersion`, `.withProtocolAndMetadata`, .`build` .. i don't think so? let me know if I'm missing anything.

Reply:
Nope will remove!